chore(suite): move from NextJS to custom build pipeline (#3446)
trezor/trezor-suite: docs/misc/build.md

> # Front-End Build

Reviewer: I'd be happier if this was placed in `packages/suite-build/README.md` but I'll leave that on you. .)

The front-end build of Suite is handled by Webpack configurations inside the `suite-build` package.

The folder structure is as follows:

> - configs: Contains the Webpack configuration files. `base.webpack.config.ts` serves as a comon base for all other configurations. The other files in this folder are project specific such as `web.webpack.config.ts` or `desktop.webpack.config.ts` for `suite-web` and `suite-desktop` respectively. 

Reviewer: I think it would be nicer for the config files to be placed in their respective packages and only the shared config to be pulled out either to the root of the repo or to this `suite-build` package.

In my ideal world each package is self-contained except dependencies and artefacts inherited from higher level directories and there are no 'side cars'.

> - configs: Contains the Webpack configuration files. `base.webpack.config.ts` serves as a comon base for all other configurations. The other files in this folder are project specific such as `web.webpack.config.ts` or `desktop.webpack.config.ts` for `suite-web` and `suite-desktop` respectively. 

Reviewer 2: ```suggestion
- configs: Contains the Webpack configuration files. `base.webpack.config.ts` serves as a common base for all other configurations. The other files in this folder are project specific such as `web.webpack.config.ts` or `desktop.webpack.config.ts` for `suite-web` and `suite-desktop` respectively. 
```

Author: cf1975a

- plugins: Contains custom Webpack plugins.
- utils: Contains various utils for the build scripts.

These Webpack configurations are using TypeScript and use the `tsconfig.json` file at the root of the package. This is specificed via the `TS_NODE_PROJECT` environment variable to avoid any issues regardless of the location where the command is run.

The following commands are available in this package (using `yarn run` at the root of the package or `yarn workspace @trezor/suite-build run` at the root of the project):
| Command | Description |
| --- | --- |

> | dev:web | Runs a watch build of `suite-web` with development settings and servies it. |

Reviewer: ```suggestion
| dev:web | Runs a watch build of `suite-web` with development settings and serves it. |
```

Author: cf1975a

> | build:web | Runs a production build of `suite-web`. |

Reviewer: ```suggestion
| build:web | Builds a production version of `suite-web`. |
```

Author: cf1975a

> | dev:desktop | Runs a watch build of `suite-desktop` with development settings, servies it and runs the Electron wrapper. |

Reviewer: ```suggestion
| dev:desktop | Runs a watch build of `suite-desktop` with development settings, serves it and runs the Electron wrapper. |
```

Also what is the electron wrapper? This makes it sound like it's an independent proces of the actual app...?

Author: cf1975a

> | build:desktop | Runs a production build of `suite-desktop`. |

Reviewer: ```suggestion
| build:desktop | Builds a production version of `suite-desktop`. |
```

Author: cf1975a
